src/main/java/software/amazon/cloudformation/AbstractWrapper.java

```diff
@@ -151,10 +151,14 @@ public AbstractWrapper(final CredentialsProvider providerCredentialsProvider,
         // Both are required parameters when LoggingConfig (optional) is provided when
         // 'RegisterType'.
         if (providerCredentials != null) {
```

Comment:
the only concern i think is, if we want to reset it to null,  we will not allow it right?

I can see three scenarios and curios on the third one
1. initially it was null, second request we got a non null provider credential we refresh it - it will work
2. initially it was non-null provider-credential, second request we got a new non-null provider credential - it will work for this case too
3. initially it was non-null provider-credenial, second request we got a null provider credential - We will still use the old one?